cuckoo/common/mongo.py

```diff
@@ -45,4 +47,161 @@ def connect(self):
+
+    def search_key(self, value):
+        return self.db.analysis.find({"behavior.summary.keys": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_mutex(self, value):
+        return self.db.analysis.find({"behavior.summary.mutex": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_domain(self, value):
+        return self.db.analysis.find({"network.domains.domain": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_ip(self, value):
+        return self.db.analysis.find({"network.hosts": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_signature(self, value):
+        return self.db.analysis.find({ "$or": [{"signatures.families": {"$regex": value, "$options": "-i"}},
+                                               {"signatures.name": {"$regex": value, "$options": "-i"}},
+                                               {"signatures.marks.call.api": {"$regex": value, "$options": "-i"}},
+                                               {"signatures.description": {"$regex": value, "$options": "-i"}}]}).sort([["_id", -1]])
+
+    def search_url(self, value):
+        return self.db.analysis.find({ "$or": [{"target.url": {"$regex": value, "$options": "-i"}},
+                                               {"target.file.urls": {"$regex": value, "$options": "-i"}}]}).sort([["_id", -1]])
+
+    def search_imphash(self, value):
+        return self.db.analysis.find({"static.pe_imphash": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_md5(self, value):
+        return self.db.analysis.find({"target.file.md5": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_sha1(self, value):
+        return self.db.analysis.find({"target.file.sha1": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_sha256(self, value):
+        return self.db.analysis.find({"target.file.sha256": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_sha512(self, value):
+        return self.db.analysis.find({"target.file.sha512": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_process_args(self, value):
+        return self.db.analysis.find({"behavior.processes.command_line": {"$regex": value, "$options": "-i"}}).sort([["_id", -1]])
+
+    def search_regkey_read(self, value):
+        return self.db.analysis.find({"behavior.summary.regkey_read": {"$elemMatch": {"$regex": value, "$options": "-i"}}}).sort([["_id", -1]])
+
+    def search_regkey_opened(self, value):
+        return self.db.analysis.find({"behavior.summary.regkey_opened": {"$elemMatch": {"$regex": value, "$options": "-i"}}}).sort([["_id", -1]])
+
+    def search_regkey_written(self, value):
+        return self.db.analysis.find({"behavior.summary.regkey_written": {"$elemMatch": {"$regex": value, "$options": "-i"}}}).sort([["_id", -1]])
+
+    def extract_database(self, results):
```

Comment:
The `mongo` module shouldn't be doing anything with our SQL database. This behavior should be moved to the `search` module, if found useful.